src/AIAPI/MainGame/GameApi.cs

```diff
@@ -0,0 +1,298 @@
+                .Select(registration => registration.Key).FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Get the first controller of the specified type that was registered with the specified extendedDataId. The type has to be an exact match.
+        /// </summary>
+        public static GameCustomFunctionController GetRegisteredBehaviour(Type controllerType, string extendedDataId)
+        {
+            return _registeredHandlers
+                .Where(registration => string.Equals(registration.Value, extendedDataId, StringComparison.Ordinal) &&
+                                       registration.Key.GetType() == controllerType)
+                .Select(registration => registration.Key).FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Register new functionality that will be added to main game. Offers easy API for custom main game logic.
+        /// All you have to do is create a type that inherits from <see cref="GameCustomFunctionController"/>>
+        /// (don't make instances, the API will make them for you). Warning: The custom controller is immediately
+        /// created when it's registered, but its OnGameLoad method is not called until a game actually loads.
+        /// This might mean that if the registration happens too late you will potentially miss some load events.
+        /// </summary>
+        /// <typeparam name="T">Type with your custom logic to add to a character</typeparam>
+        /// <param name="extendedDataId">Extended data ID used by this behaviour. Set to null if not used.</param>
+        public static void RegisterExtraBehaviour<T>(string extendedDataId) where T : GameCustomFunctionController, new()
+        {
+            if (StudioAPI.InsideStudio) return;
+
+            var newBehaviour = _functionControllerContainer.AddComponent<T>();
+            newBehaviour.ExtendedDataId = extendedDataId;
+            _registeredHandlers.Add(newBehaviour, extendedDataId);
+        }
+
+        internal static void Init(bool insideStudio)
+        {
+            if (insideStudio) return;
+
+            Hooks.SetupHooks();
+
+            _functionControllerContainer = new GameObject("GameCustomFunctionController Zoo");
+            _functionControllerContainer.transform.SetParent(Chainloader.ManagerObject.transform, false);
+
+            SceneManager.activeSceneChanged += (scene1, scene2) =>
+            {
+                var index = _newGameDetectionScenes.IndexOf(scene2.name);
+                // detect forward and backward navigation
+                if (index != -1 && scene1.name.IsNullOrWhiteSpace() && (
+                    index == _newGameDetectionIndex + 1 || index == _newGameDetectionIndex - 1))
+                {
+                    _newGameDetectionIndex = index;
+                    if (_newGameDetectionIndex + 1 == _newGameDetectionScenes.Count)
+                    {
+                        _newGameDetectionIndex = -1;
+                        OnNewGame();
```

Comment:
Not sure this applies to AI?

Reply:
This should be possible to be replaced with a hook, assuming it's needed at all. In KK this is used to ensure that controllers get a chance to run code because otherwise they can only catch savegame load and onnight events which don't fire when starting a new game.

Reply:
I took a quick look, and I think this would work:

Postfix on `TitleSceneLoad.SetWorldData` which sets a flag:

```
IsNewGame = !Manager.Game.Instance.Data.WorldList.Values.Contains(_worldData)
```

The postfix on AIProjectScene.MapScene.OnLoaded:
```
if (IsNewGame) OnNewGame();
IsNewGame = false;
```

Reply:
That was helpful.  Ended up going with `_worldData?.SaveTime.Millisecond == 0;` to detect new game using your hooks